### PR TITLE
Update solution to bob

### PR DESCRIPTION
**Pull request by aaronjpitty**
Add checks for each condition and the relevant return based on the remark

### CODE REVIEW

File: bob/bob.go [hunk 1]
```diff
@@ -1,15 +1,37 @@
...
+
+// Hey function which responds correctly
 func Hey(remark string) string {
-	// Write some code here to pass the test suite.
-	// Then remove all the stock comments.
-	// They're here to help you get started but they only clutter a finished solution.
-	// If you leave them in, reviewers may protest!
-	return ""
-}
+	remark = strings.Trim(remark, " \t\n\r")
```
Comment (danmrichards) [inline bob/bob.go]: You can actually use the `strings.TrimSpace` function here. `TrimSpace` determines the white space characters by iterating over the string looking for ASCII non-space characters. What this means is that it treats many things as whitespace, including ` `, `\t`, `\n` and `\r`

File: bob/bob.go [hunk 2]
```diff
@@ -1,15 +1,37 @@
...
+
+	re, _ := regexp.Compile("[a-zA-Z]+")
```
Comment (danmrichards) [inline bob/bob.go]: With regex it's generally good practice not to compile them at runtime. The reasons for this being that they're not particularly quick to compile and also it's not something that is likely to change.

Have a look at the `regexp.MustCompile` function, this is something that you can call outside of function iteself, using it to declare a package level variable. When you've done this compare the benchmark resuult with what you had before.

Comment (danmrichards) [reply bob/bob.go]: As an extension to this you might also want to look at the `MatchString` method in the regexp package, this is a method available on a compiled regexp object. Again checkout the benchmark results vs using `FindAllString`

File: bob/bob.go [hunk 3]
```diff
@@ -1,15 +1,37 @@
...
+
+	if question {
+		return "Sure."
+	}
+
+	if remark == "" {
+		return "Fine. Be that way!"
+	}
+
+	return "Whatever."
```
Comment (danmrichards) [inline bob/bob.go]: Have a look at the `switch` statement (https://github.com/golang/go/wiki/Switch), you will probably find you can tidy up these conditionals a little.

**Review by danmrichards — DISMISSED** [review]

File: bob/bob.go (hunk 1 above)
Comment (aaronjpitty) [reply bob/bob.go]: Oh wow thats awesome.

File: bob/bob.go [hunk 4]
```diff
@@ -1,15 +1,33 @@
...
+    case yelled:
+        return "Whoa, chill out!"
+    case question:
+        return "Sure."
+    case str == "":
+        return "Fine. Be that way!"
+    default:
+        return "Whatever."
+    }
+
```
Comment (danmrichards) [inline bob/bob.go]: The formatting looks a little messed up here. Look into running the file through `gofmt` or `goimports`.

Lots of IDE's have plugins that'll do this on save for you.

**Review by danmrichards — CHANGES_REQUESTED** [review]

File: bob/bob.go (hunk 4 above)
Comment (aaronjpitty) [reply bob/bob.go]: Yeah I'm not sure why, I had a look at it again in my VS code and it looks fine so I was a bit confused as to why git has formatted it badly. :/ 

I know Josie mentioned VSCode isn't great, but its the only editor I use 😂

Comment (aaronjpitty) [reply bob/bob.go]: Ah looks like go fmt fixed it :D

**Review by danmrichards — APPROVED** [review]
Nice, looks good!

**Pull request merged**